crates/subspace-networking/src/shared.rs

```diff
@@ -131,8 +140,11 @@ impl PreimageIntoKeyBytes<PeerId> for IdendityHash {
 }
 
 impl PreimageIntoKeyBytes<record::Key> for IdendityHash {
-    fn preimage_into_key_bytes(_record_key: &record::Key) -> KeyBytes<Self> {
-        panic!("Unsupported operation: record::Key to KeyBytes");
+    fn preimage_into_key_bytes(record_key: &record::Key) -> KeyBytes<Self> {
+        let multihash = Multihash::from_bytes(&record_key.to_vec())
+            .expect("Not multihash record keys not supported.");
```

Comment:
`.expect()` must convince reader why this this will never ever happen or why crashing an app is the best option here. The fact that only multihashes are supported is unfortunate, but I don't think it is sufficient for killing an app.

Reply:
Changed to the default SHA256 converter with a code comment.